### PR TITLE
Migrate `src/runtime/entitlements-manager` to TypeScript

### PR DESCRIPTION
**Pull request by ChrisAntaki**

### CODE REVIEW

File: src/api/entitlements.ts [hunk 1]
```diff
@@ -231,9 +231,11 @@ export class Entitlements {
    * When a metering entitlement is consumed, SwG shows the user a metering dialog.
    * When the user closes the dialog, SwG depletes one of the user's remaining
    * "free reads".
-   * @param {?Function=} onCloseDialog Called after the user closes the dialog.
    */
-  consume(onCloseDialog?: Function | null) {
+  consume(
+    /** Called after the user closes the dialog. */
```
Comment (fikus) [inline src/api/entitlements.ts:236]: This comment should be a param in the JSDoc.

File: src/runtime/entitlements-manager.ts [hunk 2]
```diff
@@ -824,24 +732,19 @@ export class EntitlementsManager {
...
 
-  /**
-   * @param {!Entitlements} entitlements
-   * @param {?Function=} onCloseDialog Called after the user closes the dialog.
-   * @private
-   */
-  consume_(entitlements, onCloseDialog) {
+  private consume_(
+    entitlements: Entitlements,
+    /** Called after the user closes the dialog. */
```
Comment (fikus) [inline src/runtime/entitlements-manager.ts:743]: Should be JSDoc param.

File: src/runtime/entitlements-manager.ts [hunk 3]
```diff
@@ -850,25 +753,30 @@ export class EntitlementsManager {
...
         // Ignore decoding errors. Don't show a toast, and return
         // onConsumeCallback directly.
         return onConsumeCallback();
       }
 
-      if (
-        entitlement.subscriptionTokenContents['metering'] &&
-        entitlement.subscriptionTokenContents['metering']['showToast'] === true
-      ) {
+      if (subscriptionTokenContents['metering']['showToast'] === true) {
```
Comment (fikus) [inline src/runtime/entitlements-manager.ts:773]: This is losing the check that ['metering'] exists, should we use optional chaining here?

**Review by fikus — APPROVED** [review]

File: src/api/entitlements.ts (hunk 1 above)
Comment (ChrisAntaki) [reply src/api/entitlements.ts]: Updated

File: src/runtime/entitlements-manager.ts (hunk 3 above)
Comment (ChrisAntaki) [reply src/runtime/entitlements-manager.ts]: Nice catch, fixed

**Pull request merged**